dvc/repo/__init__.py

```diff
@@ -21,14 +20,20 @@
 from dvc.utils.compat import open as _open, fspath_py35, FileNotFoundError
 from dvc.utils import relpath
 
+from .graph import check_acyclic, get_pipeline, get_pipelines, get_stages
+
+
 logger = logging.getLogger(__name__)
 
 
 def locked(f):
     @wraps(f)
     def wrapper(repo, *args, **kwargs):
         with repo.lock:
-            return f(repo, *args, **kwargs)
+            ret = f(repo, *args, **kwargs)
+            # Our graph cache is no longer valid after we release the repo.lock
```

Comment:
Why it's not valid? I would write that concurrent dvc process might change stage files and this graph cache might be stale.